presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestReentrantBoundedExecutor.java

```diff
@@ -31,25 +32,31 @@ public void testReentrantBoundedExecutor()
     {
         AtomicInteger callCounter = new AtomicInteger();
         SettableFuture<Object> future = SettableFuture.create();
-        Executor reentrantExecutor = new ReentrantBoundedExecutor(newCachedThreadPool(), 1);
-        reentrantExecutor.execute(() -> {
-            callCounter.incrementAndGet();
+        ExecutorService executor = newCachedThreadPool();
+        try {
+            Executor reentrantExecutor = new ReentrantBoundedExecutor(executor, 1);
             reentrantExecutor.execute(() -> {
                 callCounter.incrementAndGet();
-                future.set(null);
+                reentrantExecutor.execute(() -> {
+                    callCounter.incrementAndGet();
+                    future.set(null);
+                });
+                try {
+                    future.get();
+                }
+                catch (Exception ignored) {
```

Comment:
why exception is swallowed?

Reply:
pre-existing, i have only added try-finally here.
would need to ask test's author, @sopel39

Reply:
It's settable future used only in test. It cannot throw